core/unix/src/TUnixSystem.cxx

```diff
@@ -3582,6 +3583,7 @@ static struct Signalmap_t {
    { SIGSYS,   0, 0, "bad argument to system call" },
    { SIGPIPE,  0, 0, "write on a pipe with no one to read it" },
    { SIGILL,   0, 0, "illegal instruction" },
+   { SIGABRT,  0, 0, "abort" },
```

Comment:
What's the reasoning/purpose of this addition?  Note that "too-often" the signal handler itself gets silly (for example on my MacOS it is common for the stack trace generator to get stuck) and I need to force the end of the process with kill -9  ... does that prevent it .. or not?

Reply:
Even if I do not know how the whole signal capturing / exception mechanism works in ROOT, my guess is that this is added to handle sigabrt in case a library has a critical error and sends the signal (@wlav can better comment perhaps). Anyway I understand this should not affect sigkill.

Reply:
SIGABRT comes from assertion failures. Having the handle, cppyy can trace the abort from a C++ function in a (automatically generated) callback through Python, yielding both a C++ and Python stack trace. Not sure about the technical term "gets silly". Actually, in cppyy when using controlled execution with `with`, cppyy will by default switch the C++ tracing of. It's never been good on Mac or Windows (with the latter now changing for the better, I see) and it's dog-slow on Linux plus annoyingly duplicated. (On the wishlist to patch.)

Either which way, since the signal map is not accessible through an API, it is repeated in clingwrapper.cxx, so just as a heads up you need to patch that in your copy if you don't add SIGABRT. (Alternatively, SIGABRT could be moved to the end of the map, so then you don't have to, as it will never appear nor bump the number of the others.)

Reply:
> Not sure about the technical term "gets silly"

In my case, takes for ever and no longer react to CTRL+C or CTRL+Z, usually because something is going awry (infinite loop?, crash?) in the stack trace generator (gdb).

> (On the wishlist to patch.)

Note the related PR: (On the wishlist to patch.)

> SIGABRT comes from assertion failures. 

humm ... so my point is moot, I mistoke SIGABRT (6) for SIGKILL (9) .. so never mind :)

Reply:
CTRL+\ is your friend. It sends SIGQUIT, which the application can not intercept (contrary to SIGINT or SIGTSTP as coming from the other two).